include/nbl/system/IWindowWin32.h

```diff
@@ -0,0 +1,29 @@
+#ifndef __NBL_I_WINDOW_WIN32_H_INCLUDED__
+#define __NBL_I_WINDOW_WIN32_H_INCLUDED__
+
+#include "nbl/system/IWindow.h"
+
+#define WIN32_LEAN_AND_MEAN
+#include <windows.h>
```

Comment:
https://stackoverflow.com/questions/4111899/is-there-an-easy-one-shot-solution-to-the-preprocessor-namespace-pollution-intro

If the only reason you need this is because of `HWND` then just make `native_handle_t = void*` because HWND is just that apparently.